engines/ags/shared/util/file_stream.cpp

```diff
@@ -204,15 +206,28 @@ void FileStream::Open(const String &file_name, FileOpenMode open_mode, FileWorkM
 			else if (fname.findFirstOf('/') != Common::String::npos)
 				error("Invalid attempt to create file - %s", fname.c_str());
 
-			_outSave = g_system->getSavefileManager()->openForSaving(fname, false);
+			filename = fname;
 		}
 
+		_outSave = g_system->getSavefileManager()->openForSaving(filename, false);
 		if (!_outSave)
 			error("Invalid attempt to create file - %s", file_name.GetCStr());
 
-		// Any data written has to first go through the memory stream buffer,
-		// since the savegame code uses Seeks, which OutSaveFile doesn't support
-		_file = &_writeBuffer;
+		// Check if the save file supports seek/size.. seek is needed
+		// for backtracking to write out chunk sizes and screenshot offset
+		if (_outSave->size() == 0) {
+			// Supports it, so data can be written directly
+			_file = _outSave;
+			_outSave = nullptr;
+		} else {
+			// It doesn't. So data needs to be written to a memory buffer
+			// to allow for seeking, and will be dumped all at once at the end
+			if (::AGS::g_vm->getGameId() == "strangeland")
+				error("Your platform doesn't support savegame seeking, " \
```

Comment:
Shouldn't this be made user-visible, e.g. `GUI::MessageBox()` and translateable? If it is the latter, POTFILES needs to be modified too.